src/tribler/core/components/libtorrent/download_manager/download_state.py

```diff
@@ -156,8 +156,12 @@ def total_payload_download(self) -> int:
     def get_all_time_ratio(self) -> float:
         """ Returns the accumulated seeding ratio of the download across multiple sessions.
         """
-        if not self.lt_status or not self.all_time_download:
+        if not self.lt_status:
             return 0
+
+        if not self.all_time_download:
+            return 0 if not self.all_time_upload else float("inf")
```

Comment:
**nitpick:** I slightly prefer `math.inf` to `float("inf")`, but both are possible.

Reply:
Agree. Changed.